docs/further_useful_features.rst

```diff
@@ -0,0 +1,459 @@
+    updated_df
+
+
+The way dataset updates work is that new partitions are added to a dataset
+as long as they have the same tables as the existing partitions. A `different`
+table **cannot** be introduced into an existing dataset with an update.
+
+To illustrate this point better, let's first create a dataset with two tables:
+
+.. ipython:: python
+
+    df2 = pd.DataFrame(
+        {
+            "G": "foo",
+            "H": pd.Categorical(["test", "train", "test", "train"]),
+            "I": np.array([9] * 4, dtype="int32"),
+            "J": pd.Series(3, index=list(range(4)), dtype="float32"),
+            "K": pd.Timestamp("20190604"),
+            "L": 2.0,
+        }
+    )
+    df2
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "another_unique_dataset_identifier",
+        dfs=[{"data": {"table1": df, "table2": df2}}],
+    )
+    dm.tables
+    sorted(dm.partitions.keys())
+
+
+.. admonition:: Filenames
+
+   In the previous example a dictionary was used to pass the desired data to the store function. To label each file,
+   by default ``kartothek`` uses UUIDs to ensure that each filename is unique, this is necessary so that the
+   update can properly work using `copy-on-write <https://en.wikipedia.org/wiki/Copy-on-write>`_ principles.
+
+Below is an example where we update the existing dataset ``another_unique_dataset_identifier``
+with new data for ``table1`` and ``table2``:
+
+.. ipython:: python
+
+    another_df2 = pd.DataFrame(
+        {
+            "G": "bar",
+            "H": pd.Categorical(["prod", "dev", "prod", "dev"]),
+            "I": np.array([12] * 4, dtype="int32"),
+            "J": pd.Series(4, index=list(range(4)), dtype="float32"),
+            "K": pd.Timestamp("20190614"),
+            "L": 10.0,
+        }
+    )
+    another_df2
+
+    dm = update_dataset_from_dataframes(
+        {"data": {"table1": another_df, "table2": another_df2}},
+        store=store_factory,
+        dataset_uuid="another_unique_dataset_identifier",
+    )
+    dm.tables
+    sorted(dm.partitions.keys())
+
+
+Trying to update only a subset of tables throws a ``ValueError``:
+
+.. ipython::
+
+    @verbatim
+    In [45]: update_dataset_from_dataframes(
+       ....:        {
+       ....:           "data":
+       ....:           {
+       ....:              "table2": another_df2
+       ....:           }
+       ....:        },
+       ....:        store=store_factory,
+       ....:        dataset_uuid="another_unique_dataset_identifier"
+       ....:        )
+       ....:
+    ---------------------------------------------------------------------------
+    ValueError: Input partitions for update have different tables than dataset:
+    Input partition tables: {'table2'}
+    Tables of existing dataset: ['table1', 'table2']
+
+
+Adding data to an existing dataset is not the only functionality achievable within an update
```

Comment:
I'd restructure this chapter into subsections. 

# Dataset updates
## Append data
## Delete data
## Replace data

Feel free to choose better titles

Reply:
The replace section would be the usage of both *append* and *delete_scope* simultaneously